File: C/PCA9537.c
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// PCA9537
// This code is designed to work with the PCA9537_I2CO4_IRFR3710 I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/products

#include <stdio.h>
#include <stdlib.h>
#include <linux/i2c-dev.h>
#include <sys/ioctl.h>
#include <fcntl.h>
#include <math.h>

void main()
{
	// Create I2C bus
	int file;
	char *bus = "/dev/i2c-1";
	if((file = open(bus, O_RDWR)) < 0)
	{
		printf("Failed to open the bus. \n");
		exit(1);
	}
	// Get I2C device, PCA9537 I2C address is 0x41(65)
	ioctl(file, I2C_SLAVE, 0x41);

	// Select configuration register(0x03)
	// All pins configured as inputs(0xFF)
	char config[2] = {0};
	config[0] = 0x03;
	config[1] = 0xFF;
	if(write(file, config, 2) != 2)
	{
		printf("Error : Input/Output error \n");
		exit(1);
	}

	// Output to screen
	printf("All Pins State are HIGH \n");
	sleep(1);

	// Read 1 byte of data
	char data[1] = {0};
	if(read(file, data, 1) != 1)
	{
		printf("Error : Input/output Error \n");
	}
	else
	{
		// Output to screen
		int data1 = (data[0] & 0x0F);
		
		for(int i=0; i<4; i++)
		{
			if((data1 & ((int)Math.pow(2, i))) == 0)
			{
				printf("I/O Pin %d State is LOW \n", i);
			}
			else
			{
				printf("I/O Pin %d State is HIGH \n", i);
				sleep(0.5);
			}
		}
	}
}

File: Onion_Omega_Python/PCA9537.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# PCA9537
# This code is designed to work with the PCA9537_I2CO4_IRFR3710 I2C Mini Module available from ControlEverything.com.
# https://www.controleverything.com/products

from OmegaExpansion import onionI2C
import time

# Get I2C bus
i2c = onionI2C.OnionI2C()

# PCA9537 address, 0x41(65)
# Select configuration register, 0x03(03)
#		0xFF(255)	All pins configured as inputs
i2c.writeByte(0x41, 0x03, 0xFF)

# Output to screen
print "All Pins State are HIGH"

time.sleep(0.5)

# PCA9537 address, 0x41(65)
# Read data back from 0x00(00), 1 byte
data = i2c.readBytes(0x41, 0x00, 1)

# Convert the data to 4-bits
data = (data & 0x0F)

for i in range(0, 4) :
	if (data & (2 ** i)) == 0 :
		print "I/O Pin %d State is LOW" %i
	else :
		print "I/O Pin %d State is HIGH" %i
		time.sleep(0.5)
